lib/mayaUsd/ufe/private/Utils.cpp

```diff
@@ -159,22 +162,10 @@ void applyCommandRestriction(const UsdPrim& prim, const std::string& commandName
                 break;
             }
         }
-
-        if(!layerDisplayName.empty())
-        {
-            layerDisplayName.pop_back();
-            std::string err = TfStringPrintf("Cannot %s [%s]. It is defined on another layer. Please set %s as the target layer to proceed.",
-                                     commandName.c_str(),
-                                     prim.GetName().GetString().c_str(), 
-                                     layerDisplayName.c_str());
-            throw std::runtime_error(err.c_str());
-        }
     }
-    else
+
+    if(!layerDisplayName.empty())
```

Comment:
Simplified the logic by removing the redundant primSpec check and loop through primStack.